### DISCUSSION TITLE
How to add annotation in the `shell` code block after the `\` char?

### DISCUSSION THREAD

**Discussion by hellt** [squidfunk/mkdocs-material#6588] [Q&A] [answered]
Hi all,
When dealing with multiline bash (or other sh compatible lang) I am struggling to find a way to put an annotation to the end of the line.

Consider the following example:

```bash
someCmd --with-arg1 value1 \
        --with-arg2 value2
```

When adding the annotation to the first line like this:

```bash
someCmd --with-arg1 value1 \ #(1)!
        --with-arg2 value2
```

the extra space char is rendered after the `\` char, making the command invalid when pasted.

If I'd use `\#(1)!` then expectedly the annotation won't be recognized as the lexer won't treat this as a comment.

Appreciate any guidance on how to make it work =)

**Comment by squidfunk** [comment]
IMHO, this is essentially a problem with the lexer, because the comment will be preceded by a whitespace that it outside of the lexed comment. We've also discussed this before with more insights (and possible solutions), so you may try to use GitHub search to uncover the issue and discussion ☺️

**Reply by hellt** [reply, marked as answer]
gotcha, found the original issue and tried the suggested workaround.
With a minor modification it worked beautifully - https://github.com/squidfunk/mkdocs-material/issues/3846#issuecomment-1876914539
